src/execution/execute.ts

```diff
@@ -561,36 +604,44 @@ function executeFields(
         sourceValue,
         fieldGroup,
         fieldPath,
-        incrementalDataRecord,
+        errors,
         deferMap,
       );
 
       if (result !== undefined) {
-        results[responseName] = result;
         if (isPromise(result)) {
+          results[responseName] = result.then((resolved) => {
+            appendNewIncrementalDataRecords(acc, resolved[1]);
```

Comment:
is it faster to push each new incrementalDataRecord onto `acc[1]` here instead of using `appendNewIncrementalDataRecords`, which creates a new array each time?

Reply:
Sounds good! 74d092ef1